eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1709,6 +1726,31 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
       if (d1.localCommitPublished.exists(_.commitTx.txid == tx.txid)) {
         context.system.eventStream.publish(LocalCommitConfirmed(self, remoteNodeId, d.channelId, blockHeight + d.commitments.params.remoteParams.toSelfDelay.toInt))
       }
+      // if the local or remote commitment tx just got confirmed, we abandon anchor transactions that were created based
+      // on the other commitment: they will never confirm so we must free their wallet inputs.
+      if (d1.localCommitPublished.exists(_.commitTx.txid == tx.txid)) {
+        nodeParams.db.audit.listPublished(d.channelId).collect {
+          case tx if tx.desc == "remote-anchor" =>
+            log.warning("abandoning remote-anchor txId={} (local commit was confirmed)", tx.txId)
+            wallet.abandon(tx.txId)
+        }
+      }
+      if (d1.remoteCommitPublished.exists(_.commitTx.txid == tx.txid) || d1.nextRemoteCommitPublished.exists(_.commitTx.txid == tx.txid)) {
+        nodeParams.db.audit.listPublished(d.channelId).collect {
+          case tx if tx.desc == "local-anchor" =>
+            log.warning("abandoning local-anchor txId={} (local commit was confirmed)", tx.txId)
```

Comment:
```suggestion
            log.warning("abandoning local-anchor txId={} (remote commit was confirmed)", tx.txId)
```

Reply:
Damn, I thought I got them all right...done in https://github.com/ACINQ/eclair/pull/2818/commits/721dc0a66b470f933a7a1f92b4f5a146d9fdfdfb